lib/ts/recipe/multifactorauth/recipeImplementation.ts

```diff
@@ -437,55 +441,77 @@ export default function getRecipeInterface(
+                        const createPrimaryRes = await this.createPrimaryUser({
+                            recipeUserId: new RecipeUserId(sessionUser.id),
+                            userContext,
+                        });
+                        if (createPrimaryRes.status === "RECIPE_USER_ID_ALREADY_LINKED_WITH_PRIMARY_USER_ID_ERROR") {
+                            throw new Error("should never happen"); // new user, that's why
+                        } else if (
+                            createPrimaryRes.status ===
+                            "ACCOUNT_INFO_ALREADY_ASSOCIATED_WITH_ANOTHER_PRIMARY_USER_ID_ERROR"
+                        ) {
+                            return {
+                                status: "ACCOUNT_INFO_ALREADY_ASSOCIATED_WITH_ANOTHER_PRIMARY_USER_ID_ERROR",
+                                message: "Error setting up MFA for the user. Please contact support. (ERR_CODE_011)",
+                            };
+                        }
+                    }
 
-            if (justSignedInUserCreated) {
-                // This is a newly created user, so it must be account linked with the session user
-                if (!sessionUser.isPrimaryUser) {
-                    const createPrimaryRes = await this.createPrimaryUser({
-                        recipeUserId: new RecipeUserId(sessionUser.id),
+                    const linkRes = await this.linkAccounts({
+                        recipeUserId: justCompletedFactorUserInfo.recipeUserId,
+                        primaryUserId: sessionUser.id,
                         userContext,
                     });
-                    if (createPrimaryRes.status === "RECIPE_USER_ID_ALREADY_LINKED_WITH_PRIMARY_USER_ID_ERROR") {
-                        throw new Error("should never happen"); // new user, that's why
+
+                    if (linkRes.status === "RECIPE_USER_ID_ALREADY_LINKED_WITH_ANOTHER_PRIMARY_USER_ID_ERROR") {
+                        throw new Error("should never happen"); // new user shouldn't have this issue
+                    } else if (linkRes.status === "INPUT_USER_IS_NOT_A_PRIMARY_USER") {
+                        throw new Error("should never happen");
                     } else if (
-                        createPrimaryRes.status === "ACCOUNT_INFO_ALREADY_ASSOCIATED_WITH_ANOTHER_PRIMARY_USER_ID_ERROR"
+                        linkRes.status === "ACCOUNT_INFO_ALREADY_ASSOCIATED_WITH_ANOTHER_PRIMARY_USER_ID_ERROR"
                     ) {
                         return {
                             status: "ACCOUNT_INFO_ALREADY_ASSOCIATED_WITH_ANOTHER_PRIMARY_USER_ID_ERROR",
-                            message: "Error setting up MFA for the user. Please contact support. (ERR_CODE_011)",
+                            message:
+                                "Cannot complete factor setup as the account info is already associated with another primary user. Please contact support. (ERR_CODE_012)",
```

Comment:
Also, we need a list of the new error codes and a related description including the actions the support needs to take to fix them. (not sure if it already exists or not)

Reply:
Could you update the FDI with the new status codes? (Moving it here since I've resolved the other thread)